src/classes/STG_PanelRD_CTRL.cls

```diff
@@ -94,39 +84,79 @@ public with sharing class STG_PanelRD_CTRL extends STG_Panel {
+        delete [select id from Opportunity
+               where CloseDate > :limitDate
+               and isClosed != true
                and npe03__Recurring_Donation__r.npe03__Open_Ended_Status__c = :openLabel];
-               
+
         map<Id, npe03__Recurring_Donation__c> mapIdRD = new map<Id, npe03__Recurring_Donation__c>([select Id from npe03__Recurring_Donation__c where npe03__Open_Ended_Status__c = :openLabel]);
-        RD_RecurringDonations.updateRecurringDonationOnOppChangeFuture(mapIdRD.keySet());               
+        RD_RecurringDonations.updateRecurringDonationOnOppChangeFuture(mapIdRD.keySet());
     }
-        
+
     /*********************************************************************************************************
-    * @description The SelectOption list of RD Close Options 
+    * @description set original forecast months for legacy RDs, if added months, start the batch job
+    * @return void
+    */
+    private void setOriginalForecastMonthsForLegacyRD() {
+        if (isRD2Enabled) {
+            return;
+        }
+        //if we reduced the number of forecast months, we need to trim opps
+        if (originalForecastMonths > STG_Panel.stgService.stgRD.npe03__Opportunity_Forecast_Months__c){
+            trimOpportunities();
+        }
+        //if we added months, kick off the batch job to update the RDs
+        else if (originalForecastMonths < STG_Panel.stgService.stgRD.npe03__Opportunity_Forecast_Months__c){
+            STG_PanelRDBatch_CTRL rdbatch = new STG_PanelRDBatch_CTRL();
+            rdbatch.runBatch();
+        }
+        originalForecastMonths = (integer)STG_Panel.stgService.stgRD.npe03__Opportunity_Forecast_Months__c;
+
+    }
+
+    /*********************************************************************************************************
+    * @description The SelectOption list of RD Close Options
     */
     public list<SelectOption> listSORDCloseOptions {
         get {
         	if (listSORDCloseOptions == null) {
-	            listSORDCloseOptions = new list<SelectOption>();
-	            list<RD_RecurringDonations.RecurringDonationCloseOptions> CloseOptionValues = RD_RecurringDonations.RecurringDonationCloseOptions.values();       
-	            for (RD_RecurringDonations.RecurringDonationCloseOptions rdco : CloseOptionValues) {
-	                listSORDCloseOptions.add(new SelectOption(rdco.name(), rdco.name()));
-	            }
+	            listSORDCloseOptions = getSelectOptionsBasedUponLegacyOrEnhanced();
         	}
             return listSORDCloseOptions;
         }
         private set;
-    }    
+    }
+
+    /*********************************************************************************************************
+    * @description Gets the list of select options based upon Enhanced or Legacy Recurring donations
+    * return the select options of rd close options
+    */
+    private List<SelectOption> getSelectOptionsBasedUponLegacyOrEnhanced() {
```

Comment:
- Let's rename to `getCloseOptions()` or `getCloseActions()` or similar
- The description and return in the ApexDoc need more detail.
- There are obsolete vars (`CloseOptionValues` for example).
- There are vars that are simpler if renamed.

Example of the modified method:
```    
    /***
    * @description Returns close options (actions) applicable to open Opportunities on a closed RD. 
    * Options have different source depending if the enhanced Recurring Donations are enabled.
    * @return List<SelectOption> 
    */
    private List<SelectOption> getCloseOptions() {
        List<SelectOption> options = new List<SelectOption>();

        if (isRD2Enabled) {
            for (RD2_Constants.CloseActions option : RD2_Constants.CloseActions.values()) {
                options.add(new SelectOption(option.name(), option.name()));
            }
        } else {
            for (RD_RecurringDonations.RecurringDonationCloseOptions option : RD_RecurringDonations.RecurringDonationCloseOptions.values()) {
                options.add(new SelectOption(option.name(), option.name()));
            }
        }

        return options;
    }
```